docs/TOKENS.md

```diff
@@ -13,133 +13,141 @@ Spoilers. My love for Astro has grown deeper.
+The file path we will use to generate our token is `pages/api/token.json.ts`. This endpoint will return JSON data, so the file name ends with `.json.ts`. I like to put all endpoints within an `api` folder so that they are separated from front-end routes. Since the Typescript extension gets removed and the `pages` directory doesn't show up in the URL, our final endpoint will be `api/token.json`.
 
 ## Define the endpoint
-Our endpoint will be a `GET` request because our objective is to retrieve a token. You can define this in Astro by exporting a `GET` function and returning a JSON Response.
-
-We also import the `agora-token` package and bring in our `APP_ID` and `APP_CERTIFICATE`, which we defined in the `.env` file.
-
+Because our endpoint will need some input information, we will use a `POST` request and retrieve the information from the body. You can define this in Astro by exporting a `POST` function and returning a Response object.
 
 ```ts
 import type { APIContext } from "astro";
 
-export async function GET({ params }: APIContext) {
-  return new Response(
-    JSON.stringify({
-      rtcToken: "token",
-    })
-  )
-}
-```
-
-## CORS
-We first need to set up our headers so that we don't encounter any CORS issues when using our backend.
-
-```ts
-export async function GET({ params }: APIContext) {
-    const headers = new Headers({
-        "Access-Control-Allow-Origin": "*",
-        "Access-Control-Allow-Methods": "GET, OPTIONS",
-    });
+export async function POST({ request }: APIContext) {
+    const { channel, role, uid, expireTime } = await request.json()
 
     return new Response(JSON.stringify({
-        rtcToken: "token"
-    }), { headers })
+        token: "token"
+    }), { status: 200 })
 }
 ```
 
+## Response Helper Functions
+We will abstract away the Response objects in a `utils/sendResponse.ts` file to simplify our endpoint. 
+
+First, we need to set up our headers to avoid any CORS issues when using our backend.
+
 * "Access-Control-Allow-Origin": "*" allows all domains to access the server, which is helpful for APIs or during development.
-* "Access-Control-Allow-Methods": "GET, OPTIONS" specifies that only GET and OPTIONS methods are permitted, controlling how other sites can interact with the server.
+* "Access-Control-Allow-Methods": "POST, OPTIONS" specifies that only POST and OPTIONS methods are permitted, controlling how other sites can interact with the server.
 
 These headers will allow our API to work well in production and development by allowing all domains. For production, you should limit only the origins that should be allowed.
 
+Then we craft a `sendBadRequest` function with a status `400` and a `sendSuccessfulResponse` function with a status `200` that uses the headers we defined.
```

Comment:
craft sounds very robotic in this context.

Reply:
Explain what `sendBadRequest` and `sendSuccessfulResponse` do besides the 200 status. It doesn't mention the headers at all, even though they are in the code.

Reply:
It does mention the headers in that sentence and above. 

Here are updates to explain the functions better:


Then, we define two functions that create a response using the headers we defined:
1. `sendBadRequest` - which returns a status of `400` with the reason why the request was bad
2. `sendSuccessfulResponse` - which returns a status of `200` along with the data